### ISSUE TITLE
Error when viewing.

### ISSUE THREAD

**Issue by DJBoxer** [ShinDarth/TC-JSON-API#27]
Just grabbed latest release FULL.TrinityCore.JSON.REST.API.0.7 but unsure about what is wrong here.
May you take a look please. using xamp with php 7.1.4

`[2017-06-30 13:03:24] production.ERROR: ErrorException: Function mcrypt_get_iv_size() is deprecated in C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php:12241
Stack trace:
#0 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(12241): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8192, 'Function mcrypt...', 'C:\\xampp\\htdocs...', 12241, Array)
#1 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(12232): Illuminate\Encryption\Encrypter->updateBlockSize()
#2 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(6796): Illuminate\Encryption\Encrypter->setCipher('rijndael-128')
#3 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1066): Illuminate\Encryption\EncryptionServiceProvider->Illuminate\Encryption\{closure}(Object(Illuminate\Foundation\Application), Array)
#4 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1019): Illuminate\Container\Container->build(Object(Closure), Array)
#5 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1543): Illuminate\Container\Container->make('encrypter', Array)
#6 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1111): Illuminate\Foundation\Application->make('encrypter')
#7 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1095): Illuminate\Container\Container->resolveClass(Object(ReflectionParameter))
#8 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1081): Illuminate\Container\Container->getDependencies(Array, Array)
#9 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1019): Illuminate\Container\Container->build('Illuminate\\Cook...', Array)
#10 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1543): Illuminate\Container\Container->make('Illuminate\\Cook...', Array)
#11 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(9236): Illuminate\Foundation\Application->make('Illuminate\\Cook...')
#12 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(2589): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#13 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(9236): Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#14 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#15 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(9227): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#16 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1996): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#17 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1983): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#18 C:\xampp\htdocs\TC-JSON-API\public\index.php(52): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#19 {main}  
[2017-06-30 13:03:24] production.ERROR: ErrorException: Function mcrypt_get_iv_size() is deprecated in C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php:12241
Stack trace:
#0 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(12241): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(8192, 'Function mcrypt...', 'C:\\xampp\\htdocs...', 12241, Array)
#1 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(12232): Illuminate\Encryption\Encrypter->updateBlockSize()
#2 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(6796): Illuminate\Encryption\Encrypter->setCipher('rijndael-128')
#3 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1066): Illuminate\Encryption\EncryptionServiceProvider->Illuminate\Encryption\{closure}(Object(Illuminate\Foundation\Application), Array)
#4 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1019): Illuminate\Container\Container->build(Object(Closure), Array)
#5 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1543): Illuminate\Container\Container->make('encrypter', Array)
#6 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1111): Illuminate\Foundation\Application->make('encrypter')
#7 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1095): Illuminate\Container\Container->resolveClass(Object(ReflectionParameter))
#8 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1081): Illuminate\Container\Container->getDependencies(Array, Array)
#9 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1019): Illuminate\Container\Container->build('Illuminate\\Cook...', Array)
#10 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(1543): Illuminate\Container\Container->make('Illuminate\\Cook...', Array)
#11 C:\xampp\htdocs\TC-JSON-API\vendor\compiled.php(2002): Illuminate\Foundation\Application->make('Illuminate\\Cook...')
#12 C:\xampp\htdocs\TC-JSON-API\public\index.php(60): Illuminate\Foundation\Http\Kernel->terminate(Object(Illuminate\Http\Request), Object(Symfony\Component\HttpFoundation\Response))
#13 {main}  `

**Comment by Helias** [comment]
Did you install php-mcrypt or php-mbstring?

**Comment by DJBoxer** [comment]
I have mstring in my php.ini.
`extension=php_mbstring.dll`

Looks like I don't have extension for Mcrypt? when I search in file these are the only mentions.

`[mcrypt]
; For more information about mcrypt settings see http://php.net/mcrypt-module-open

; Directory where to load mcrypt algorithms
; Default: Compiled in into libmcrypt (usually /usr/local/lib/libmcrypt)
;mcrypt.algorithms_dir=

; Directory where to load mcrypt modes
; Default: Compiled in into libmcrypt (usually /usr/local/lib/libmcrypt)
;mcrypt.modes_dir=`

just checked with php.info

![mcrypt](https://user-images.githubusercontent.com/16749874/27760233-3e70f3c4-5df7-11e7-98cb-729cd4cb5682.PNG)
![mstring](https://user-images.githubusercontent.com/16749874/27760235-4a64f5ae-5df7-11e7-9225-0be6a86ff564.PNG)

[Found this guide ](http://www.myoddweb.com/2010/11/18/install-mcrypt-for-php-on-windows/)for mcrypt as I'm using xamp and seen on TC thread another user using xamp so can be done but anyways. I tried what it said here with no luck tried for 7.1.6 and older version 5.x.
I'm not seeing not libmcrypt.dll.

Then I googled the actual error to find this everywhere I looked.


> As you have discovered, mcrypt_get_iv_size() is deprecated as of PHP 7.1.
> Upgrading to Laravel 5.1 or later should make this warning go away since it uses openssl instead of mcrypt.

**Comment by Helias** [comment]
Download the development version, cloning or downloading the repository.
Follow this guide to install the project https://github.com/ShinDarth/TC-JSON-API/blob/master/INSTALL.md

**Comment by DJBoxer** [comment]
K thanks Helias, I will give it a try was hoping for easier solution lol.

*update.

Followed the guide and I successfully installed everything. There was one small issue during your install guide. when I had to do this 

`php artisan migrate`

Wouldn't allow me until I edited .env to allow connection to MySQL. maybe add a note to install guide for other users.

Final results...
![capture](https://user-images.githubusercontent.com/16749874/27767233-e9404da6-5ea4-11e7-8c7d-cf4b53259410.PNG)

So thank you Helias !!
